test/builder/task.go

```diff
@@ -239,20 +239,30 @@ func InputsResource(name string, resourceType v1alpha1.PipelineResourceType, ops
 	}
 }
 
+func ResourceOptional(optional bool) TaskResourceOp {
```

Comment:
This is totally fine but is there a easier/simpler for doing test builders for boolean values? @vdemeester

Reply:
I dont think so, its more about following our convention of building a resource, for example, 

Required Resource:

```
tb.InputsResource("resource-to-build", v1alpha1.PipelineResourceTypeGit),
```
Optional Resource:

```
tb.InputsResource("optional-resource-to-build", v1alpha1.PipelineResourceTypeGit, tb.ResourceOptional(true)),
```

But would be ideal to refactor the way we initialize a resource in future, as we add more keys/fields, we end up adding setters for each.